GeneratorInterface/GenFilters/plugins/MCMultiParticleFilter.cc

```diff
@@ -96,6 +121,40 @@ MCMultiParticleFilter::MCMultiParticleFilter(const edm::ParameterSet& iConfig)
     status_.push_back(defstat[0]);
   while (motherID_.size() < particleID_.size())
     motherID_.push_back(defmother[0]);
+
+  // if decayRadiusMin size smaller than particleID , fill up further with defaults
+  if (particleID_.size() > decayRadiusMin.size()) {
+    std::vector<double> decayRadiusmin2;
+    for (unsigned int i = 0; i < particleID_.size(); i++) {
+      decayRadiusmin2.push_back(-10.);
+    }
+    decayRadiusMin = decayRadiusmin2;
+  }
+  // if decayRadiusMax size smaller than particleID , fill up further with defaults
+  if (particleID_.size() > decayRadiusMax.size()) {
+    std::vector<double> decayRadiusmax2;
+    for (unsigned int i = 0; i < particleID_.size(); i++) {
+      decayRadiusmax2.push_back(1.e5);
+    }
+    decayRadiusMax = decayRadiusmax2;
+  }
+
+  // if decayZMin size smaller than particleID , fill up further with defaults
+  if (particleID_.size() > decayZMin.size()) {
+    std::vector<double> decayZmin2;
+    for (unsigned int i = 0; i < particleID_.size(); i++) {
+      decayZmin2.push_back(-1.e5);
+    }
+    decayZMin = decayZmin2;
+  }
+  // if decayZMax size smaller than particleID , fill up further with defaults
+  if (particleID_.size() > decayZMax.size()) {
+    std::vector<double> decayZmax2;
+    for (unsigned int i = 0; i < particleID_.size(); i++) {
+      decayZmax2.push_back(1.e5);
+    }
+    decayZMax = decayZmax2;
+  }
```

Comment:
Doesn't that mean that you will also overwrite already existing values and not only fill up the vector.

Reply:
But wouldn't it be more suitable to just add the defaults and not replace all? This is also what the description says. The code just doesn't do it.

Reply:
Many thanks. In fact I fully agree that it is more suitable to just add the defaults on top of the existing vector –– if different implementations between two filters are allowed, I'll change this (or I should also purpose to change MCSmartSingleParticleFilter at the same time?). Please let me know.